src/wp-includes/post.php

```diff
@@ -310,6 +310,68 @@ function create_initial_post_types() {
 		)
 	);
 
+	if ( theme_supports_block_templates() ) {
```

Comment:
@youknowriad I was wondering about these checks (especially here and in `taxonomy.php` where we register the `wp_theme` taxonomy). There doesn't seem to be any precedent in Core for conditionally registering a post type or taxonomy; and I was wondering if this made sense UX-wise.

I take it that the main reason for this is to hide the Template Editor for themes that don't support block templates. But this seems to make less sense if we're not even actually using any block template files from a given theme for now -- but only `wp_template` CPTs. In other words, why should the theme decide whether or not the user can create custom templates, if it's not even providing the block template file fallback?

Reply:
> why should the theme decide whether or not the user can create custom templates, if it's not even providing the block template file fallback?

Because we want themes to have the possibility to opt-out of this feature and stay as they are today. Whether or not the CPT should always be registered or not is another concern though.

Reply:
Making CPT and taxonomy registration unconditional in 3471404 and cba7648, respectively.

Reply:
> Making CPT and taxonomy registration unconditional in 3471404 and cba7648, respectively.

👍 This is a good call, as it prevents problems with the post type and taxonomy disappearing when importing/exporting because the current theme doesn't happen to use it.